ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,41 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            if (t0 && t1)
+            {
+                bool isAllResult;
+                size32_t resultBytes;
+                rtlDataAttr matchResults;
+
+                if(isUnicodeType(t0->queryType()))
+                {
+                    size32_t plen = t0->queryType()->getStringLen();
+                    OwnedMalloc<UChar> pattern (plen+1);
+                    t0->getUCharStringValue(plen+1, pattern.get()); //plen+1 so get null-terminated
+                    size32_t slen = t1->queryType()->getStringLen();
+                    OwnedMalloc<UChar> search (slen);
+                    t1->getUCharStringValue(slen, search);
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern, !expr->hasAttribute(noCaseAtom));
+                    compiled->getMatchSet(isAllResult, resultBytes, matchResults.refdata(), slen, search.get());
+                    rtlDestroyCompiledUStrRegExpr(compiled);
+                }
+                else
+                {
+                    StringBuffer pattern, search;
+                    t0->getStringValue(pattern);
+                    t1->getStringValue(search);
+                    rtlCompiledStrRegex compiled;
+                    compiled.setPattern(pattern.str(), !expr->hasAttribute(noCaseAtom));
+                    compiled->getMatchSet(isAllResult, resultBytes, matchResults.refdata(), search.length(), search.str());
+                }
+                return convertSetToExpression(isAllResult, resultBytes, matchResults.getdata(), expr->getType());
```

Comment:
This needs to pass expr->queryType().  getType() links the type, but that isn't needed in this situation.